### ISSUE TITLE
REQUEST: Migrate cncf/apisnoop

### ISSUE THREAD

**Issue by BobyMCbobs** [kubernetes/org#4705]
### New repo, staging repo, or migrate existing

apisnoop

### Is it a staging repo?

yes

### Requested name for new repository

apisnoop

### Which Organization should it reside

kubernetes-sigs

### Who should have admin access?

hh, heyste, cooldracula, zachmandeville, BobyMCbobs

### Who should be listed as approvers in OWNERS?

hh, heyste, cooldracula, zachmandeville, BobyMCbobs

### Who should be listed in SECURITY_CONTACTS?

hh, heyste, cooldracula, zachmandeville, BobyMCbobs

### What should the repo description be?

⭕️Snooping on the Kubernetes OpenAPI communications 

### What SIG and subproject does this fall under?

sig-architecture

### Please provide references to appropriate approval for this new repo

- https://kubernetes.slack.com/archives/C78F00H99/p1683662771165129
- https://kubernetes.slack.com/archives/C78F00H99/p1683663250267499
- https://groups.google.com/u/3/g/kubernetes-sig-architecture/c/bEx9fppYQ0Q/m/8ZyumVYUBwAJ

### Additional context for request

CNCF is wanting to donate Kubernetes conformance infrastructure (cncf/apisnoop, cncf-infra/verify-conformance, [conformance-gate](https://git.k8s.io/test-infra/config/jobs/kubernetes/sig-arch/conformance-gate.yaml)) to sig-architecture and sig-k8s-infra.

**Comment by MadhavJivrajani** [comment]
/assign

I can help you out with this!

@BobyMCbobs we will need to ensure that cncf/apisnoop resembles our template repository for migration: https://github.com/kubernetes/kubernetes-template-project

Can you pls help me out with that?

**Comment by BobyMCbobs** [comment]
> /assign
> 
> I can help you out with this!

@MadhavJivrajani, thank you!
 
> @BobyMCbobs we will need to ensure that cncf/apisnoop resembles our template repository for migration: https://github.com/kubernetes/kubernetes-template-project
> 
> Can you pls help me out with that?

I've added the appropriate files to the repo

**Comment by MadhavJivrajani** [comment]
Just an update here, once @BobyMCbobs transfers cncf/apisnoop to our intermediate org `kubernetes-purgatory` (invite sent), I can take it from there and setup the relevant teams and documentation.

**Comment by MadhavJivrajani** [comment]
@BobyMCbobs 
please also fill in
> Who should have write access?

If this is the same as admin access, feel free to copy the same list over.

**Comment by BobyMCbobs** [comment]
> @BobyMCbobs please also fill in
> 
> > Who should have write access?
> 
> If this is the same as admin access, feel free to copy the same list over.

@MadhavJivrajani, done so! Thank you!

**Comment by MadhavJivrajani** [comment]
👋🏼 hey all

The repo as it is has admins that are not part of the [kubernetes-sigs](https://github.com/kubernetes-sigs) org, and we require them to be. I am going to remove them for now and they can go ahead with [applying for membership](https://github.com/kubernetes/community/blob/master/community-membership.md#member). These are the folks that would need to do that:
- @bernokl
- @devaips
- @zachmandeville (I'm not sure if they are a member because GitHub activity is private).

and also @cooldracula (since they are listed in the issue description).

I'm going to go ahead and add these folks into kubernetes-sigs since they are already part of one of the existing Kubernetes orgs (done as part of https://github.com/kubernetes/org/pull/4734):
- @heyste
- @mgdevstack
- @RobertKielty

Once the repo migration is complete and the requisite access has been granted, folks with admin and write permissions will be the ones that are listed in the issue description.

Once all of the above folks are part of k-sigs, the repo admins can then go ahead and PR whoever they think should have admin/write access into the respective GitHub teams (as done here: https://github.com/kubernetes/org/pull/4734)

**Comment by MadhavJivrajani** [comment]
We are almost there following the steps listed here: https://github.com/kubernetes/community/blob/master/github-management/kubernetes-repositories.md#rules-for-donated-repositories

@midopooler 👋🏼 

Since you have contributed to `github.com/cncf/apisnoop` (https://github.com/kubernetes-sigs/apisnoop/commit/def6ad1e246735565c5f2014a20b32495342e3ef), would you be able to sign the CNCF CLA agreement that is required for contributors committing changes to any repository in any of the Kubernetes orgs. The steps to do so can be found [here](https://github.com/kubernetes/community/blob/fe401d22f338340d0189144656faa45c12c86cb7/CLA.md).

**Comment by BobyMCbobs** [comment]
@MadhavJivrajani, thank you!
Out of all those listed above, @heyste @cooldracula and @zachmandeville are the only needed and the rest are not active contributors.

**Comment by MadhavJivrajani** [comment]
@BobyMCbobs great, thanks for clarifying - the list I mentioned was based on the admins that were part of the repo pre-migration.

@zachmandeville and @cooldracula could you please go ahead and apply for membership using the guidelines here: https://github.com/kubernetes/community/blob/master/community-membership.md#member

**Comment by BobyMCbobs** [comment]
> @BobyMCbobs great, thanks for clarifying - the list I mentioned was based on the admins that were part of the repo pre-migration.
> 
> @zachmandeville and @cooldracula could you please go ahead and apply for membership using the guidelines here: https://github.com/kubernetes/community/blob/master/community-membership.md#member

Apologies for changing it around. Just doing some coordinating and @cooldracula will go ahead with it and @zachmandeville will not be

**Comment by cooldracula** [comment]
Thank you for coordinating this.  I am arranging sponsors and then will apply for the membership!

**Comment by MadhavJivrajani** [comment]
Update:
- Org membership for @cooldracula should now be sorted out 
- Once the following 2 PRs merge, we can close this out:
  - https://github.com/kubernetes/community/pull/7723
  - https://github.com/kubernetes-sigs/apisnoop/pull/849

**Comment by MadhavJivrajani** [comment]
The above PRs have merged, this is complete now, thanks all!
/close

**Comment by BobyMCbobs** [comment]
> The above PRs have merged, this is complete now, thanks all! /close

@MadhavJivrajani, thank you kindly for your support and assistance with making this possible!
